### PR TITLE
[Go] Add simple RAG sample

### PR DESCRIPTION
**Pull request by ssbushi**
- [x] Docs updated

### CODE REVIEW

File: go/samples/rag/main.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+type simpleQaPromptInput struct {
+	Query   string `json:"query"`
+	Context string `json:"context"`
+}
+
+func main() {
+	apiKey := os.Getenv("GOOGLE_GENAI_API_KEY")
+	if apiKey == "" {
+		fmt.Fprintln(os.Stderr, "coffee-shop example requires setting GOOGLE_GENAI_API_KEY in the environment.")
```
Comment (ianlancetaylor) [inline go/samples/rag/main.go:73]: This is now the rag example, not the coffee-shop example.

File: go/samples/rag/main.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	embedder, err := googleai.NewEmbedder(context.Background(), "embedding-001", apiKey)
+	if err != nil {
+		log.Fatal(err)
+	}
+	localDb, err := localvec.New(context.Background(), "/tmp/", "simpleQa", embedder, nil)
```
Comment (ianlancetaylor) [inline go/samples/rag/main.go:98]: Use `os.TempDir()` rather than `"/tmp/"`.

File: go/samples/rag/main.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+	genkit.DefineFlow("simpleQaFlow", func(ctx context.Context, input *simpleQaInput, _ genkit.NoStream) (string, error) {
+		d1 := ai.DocumentFromText("Paris is the capital of France", nil)
+		d2 := ai.DocumentFromText("USA is the largest importer of coffee", nil)
+		d3 := ai.DocumentFromText("Water exists in 3 states - solid, liquid and gas", nil)
+
+		indexerReq := &ai.IndexerRequest{
+			Documents: []*ai.Document{d1, d2, d3},
+		}
+		localDb.Index(ctx, indexerReq)
```
Comment (ianlancetaylor) [inline go/samples/rag/main.go:111]: Check the error result.

**Review by apascal07 — APPROVED** [review]

File: go/samples/rag/main.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+		dRequest := ai.DocumentFromText(input.Question, nil)
+		retrieverReq := &ai.RetrieverRequest{
+			Document: dRequest,
+		}
+		response, err := localDb.Retrieve(ctx, retrieverReq)
+		if err != nil {
+			return "", err
+		}
+
+		var context string
```
Comment (ianlancetaylor) [inline go/samples/rag/main.go:122]: Better coding would be a `strings.Builder`.

```Go
    var sb strings.Builder
    for _, d := range response.Documents {
        sb.WriteString(d.Content[0].Text())
        sb.WriteByte('\n')
    }

    ...

        Context: sb.String(),

**Review by ianlancetaylor — APPROVED** [review]

File: go/samples/rag/main.go (hunk 1 above)
Comment (ssbushi) [reply go/samples/rag/main.go]: Done.

File: go/samples/rag/main.go (hunk 2 above)
Comment (ssbushi) [reply go/samples/rag/main.go]: Done.

File: go/samples/rag/main.go (hunk 3 above)
Comment (ssbushi) [reply go/samples/rag/main.go]: Done.

File: go/samples/rag/main.go (hunk 4 above)
Comment (ssbushi) [reply go/samples/rag/main.go]: Done.

**Pull request merged**